### PR TITLE
mgr: use the ac-user-set-password cmd to update the password

### PR DESCRIPTION
**Pull request by rkachach**
previous code was calling the following command to create the dashboard user:
> ceph dashboard ac-user-create <username> -i <password-file>
this works the first time when the user doesn't exist, however subsequent calls for example when the password has changed or when the secret was deleted from k8s, will not take place as this command will succeeded because the user already exists but the password is not updated in this case. The fix consists in adding an explicit call to the command:
> ceph dashboard ac-user-set-password <username> -i <password-file>
which ensures the password is updated correctly in the access control db

Fixes: https://github.com/rook/rook/issues/13594

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by rkachach** [comment]
@nizamial09 please, have a look and let me know in case of any comments :+1:

File: pkg/operator/ceph/cluster/mgr/dashboard.go [hunk 1]
```diff
@@ -270,19 +270,37 @@ func (c *Cluster) setLoginCredentials(password string) error {
...
 		}
 	}()
 
+	// Create dashboard user
+	// > ceph dashboard ac-user-create <username> -i <path-to-password-file>
+	//
+	// Note: this command will succeed in cast the user <dashboardUsername> already
+	// exists however it will not update the password. That why we need to explicitly
+	// call the ac-user-set-password command to ensure the password is updated correctly
+	args = []string{"dashboard", "ac-user-create", dashboardUsername, "-i", file.Name(), "administrator"}
```
Comment (parth-gr) [inline pkg/operator/ceph/cluster/mgr/dashboard.go:286]: can we just append the administrator in the arg

File: pkg/operator/ceph/cluster/mgr/dashboard.go [hunk 2]
```diff
@@ -270,19 +270,37 @@ func (c *Cluster) setLoginCredentials(password string) error {
...
+		output, err := client.NewCephCommand(c.context, c.clusterInfo, args).RunWithTimeout(exec.CephCommandsTimeout)
+		return "create dashboard user", output, err
+	}, 5, dashboardInitWaitTime)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create dashboard user %q", dashboardUsername)
+	}
+
+	// Set dashboard user password
+	// > ceph dashboard ac-user-set-password <username> -i <path-to-password-file>
+	args = []string{"dashboard", "ac-user-set-password", dashboardUsername, "-i", file.Name()}
```
Comment (parth-gr) [inline pkg/operator/ceph/cluster/mgr/dashboard.go:297]: what makes it different in rerunning

File: pkg/operator/ceph/cluster/mgr/dashboard.go [hunk 3]
```diff
@@ -270,19 +270,37 @@ func (c *Cluster) setLoginCredentials(password string) error {
...
 
+	// Create dashboard user
+	// > ceph dashboard ac-user-create <username> -i <path-to-password-file>
+	//
+	// Note: this command will succeed in cast the user <dashboardUsername> already
+	// exists however it will not update the password. That why we need to explicitly
+	// call the ac-user-set-password command to ensure the password is updated correctly
+	args = []string{"dashboard", "ac-user-create", dashboardUsername, "-i", file.Name(), "administrator"}
+	_, err = client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
+		output, err := client.NewCephCommand(c.context, c.clusterInfo, args).RunWithTimeout(exec.CephCommandsTimeout)
```
Comment (parth-gr) [inline pkg/operator/ceph/cluster/mgr/dashboard.go:288]: as this is a common method we can make a separate function for this passing different arg values

**Review by parth-gr — CHANGES_REQUESTED** [review]

File: pkg/operator/ceph/cluster/mgr/dashboard.go (hunk 2 above)
Comment (rkachach) [reply pkg/operator/ceph/cluster/mgr/dashboard.go:297]: @parth-gr it's a different command: `ac-user-set-password`

File: pkg/operator/ceph/cluster/mgr/dashboard.go (hunk 3 above)
Comment (rkachach) [reply pkg/operator/ceph/cluster/mgr/dashboard.go:288]: not sure if it has enough entity to be its own function as it's just a simple wrapper:

```
	_, err = client.ExecuteCephCommandWithRetry(func() (string, []byte, error) {
		output, err := client.NewCephCommand(c.context, c.clusterInfo, args).RunWithTimeout(exec.CephCommandsTimeout)

```

File: pkg/operator/ceph/cluster/mgr/dashboard.go (hunk 1 above)
Comment (parth-gr) [reply pkg/operator/ceph/cluster/mgr/dashboard.go:286]: I was wondering we can make a common arg array and call the function with different args

File: pkg/operator/ceph/cluster/mgr/dashboard.go (hunk 3 above)
Comment (parth-gr) [reply pkg/operator/ceph/cluster/mgr/dashboard.go:288]: okay

File: pkg/operator/ceph/cluster/mgr/dashboard.go (hunk 1 above)
Comment (rkachach) [reply pkg/operator/ceph/cluster/mgr/dashboard.go:286]: Ohh I see, thanks for the clarification. The thing is that in this case the commands are different (even when it seems like they do similar actions) so I don't see the benefit of factoring at this level.

**Review by travisn — APPROVED** [review]

**Review by parth-gr — APPROVED** [review]

**Review by nizamial09 — APPROVED** [review]
lgtm! thank you @rkachach

**Pull request merged**